packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -405,16 +407,20 @@ impl ScriptTransactionBuilder {
             .collect()
     }
 
-    fn no_spendable_input<'a, I: IntoIterator<Item = &'a FuelInput>>(inputs: I) -> bool {
-        !inputs.into_iter().any(|i| {
-            matches!(
-                i,
-                FuelInput::CoinSigned(_)
-                    | FuelInput::CoinPredicate(_)
-                    | FuelInput::MessageCoinSigned(_)
-                    | FuelInput::MessageCoinPredicate(_)
-            )
-        })
+    fn no_input_to_cover_fees<'a, I: IntoIterator<Item = &'a FuelInput>>(inputs: I) -> bool {
+        inputs
+            .into_iter()
+            .any(|i| match i {
+                FuelInput::CoinSigned(CoinSigned { asset_id, .. })
+                | FuelInput::CoinPredicate(CoinPredicate { asset_id, .. })
+                    if *asset_id == BASE_ASSET_ID =>
+                {
+                    true
+                }
+                FuelInput::MessageCoinSigned(_) | FuelInput::MessageCoinPredicate(_) => true,
+                _ => false,
+            })
+            .not()
     }
```

Comment:
Some points:
1. `no_input_to_cover_fees` implies that any base input input will be able to cover fees which is misleading. Maybe indicate that you don't care about the amount but rather that only an input of the base asset exists.
2. The fn reads weird. I start off reading the name saying "i say whether we have no inputs to cover the fee". With that in mind I proceed to read that we're asking whether we have any input that CAN cover it. That took me a few reads until I saw the `not` at the end. I suggest a more linear flow by either inverting the function name or by introducing a temp variable:

So something like this:
```rust
    fn no_base_asset_input<'a, I: IntoIterator<Item = &'a FuelInput>>(inputs: I) -> bool {
        let has_base_asset = inputs.into_iter().any(|i| match i {
            FuelInput::CoinSigned(CoinSigned { asset_id, .. })
            | FuelInput::CoinPredicate(CoinPredicate { asset_id, .. })
                if *asset_id == BASE_ASSET_ID =>
            {
                true
            }
            FuelInput::MessageCoinSigned(_) | FuelInput::MessageCoinPredicate(_) => true,
            _ => false,
        });
        !has_base_asset
    }
```